crates/autopilot/src/domain/auction/mod.rs

```diff
@@ -22,3 +22,35 @@ pub struct AuctionWithId {
     pub id: Id,
     pub auction: Auction,
 }
+
+/// The price of a token in wei. This represents how much wei is needed to buy
+/// 10**18 of another token.
+#[derive(Debug, Clone, Copy)]
+pub struct Price(pub eth::Ether);
+
+impl Price {
+    /// The base Ether amount for pricing.
+    const BASE: u128 = 10_u128.pow(18);
+
+    /// Apply this price to some token amount, converting that token into ETH.
+    ///
+    /// # Examples
+    ///
+    /// Converting 1 ETH expressed in `eth::TokenAmount` into `eth::Ether`
+    ///
+    /// ```
+    /// use autopilot::domain::{auction::Price, eth};
+    ///
+    /// let amount = eth::TokenAmount::from(eth::U256::exp10(18));
+    /// let price = Price(eth::Ether::from(eth::U256::exp10(18)));
+    ///
+    /// let eth = price.in_eth(amount);
+    /// assert_eq!(eth, eth::Ether::from(eth::U256::exp10(18)));
```

Comment:
I didn't notice this in the other codebase but this example is actually pretty bad (it's converting 1 of something priced 1 which doesn't really convey how things are working since both 1*1 as well a 1/1 is 1)

Can you change the example to use round but non-trivial values?

Reply:
Fixed.